### DISCUSSION TITLE
NOT NULL constraint failed after Laravel 11 upgrade

### DISCUSSION THREAD

**Discussion by joshembling** [laravel/framework#52182] [General]
I have a bunch of these errors happening since upgrading to Laravel 11 and figured out it is due to some migrations where the column type has changed. 

Original:
`$table->boolean('enable')->default(1);`

Updated: 
`$table->integer('enable')->change();`

It seems to lose the default value when changing the type. This has never been an issue as far as I am aware in previous versions. 

It is fixed by doing this:
`$table->integer('enable')->change()->default(1);`

Does anyone know if this is a bug or expected behaviour? I can see both points.

**Comment by ghost** [comment]
Hi. From experience, change() usually replaces the column definition except that when the type is the same, then only the specified attributes change on mysql from what I saw. Ex length for strings and the default remains the same if not specified. In mysql your change would transform the tiny int into int so I guess that is a type change.

**Reply by joshembling** [reply]
I suppose the question is then, why has this always worked in tests before now? 😂

**Reply by ghost** [reply]
Obviously something changed in the code:) When I used change and saw it does not change the default value I had to change the migration to change also the default. And at that point I saw it as a bug, (but I did not change the type). You can only go and debug the code to see the commits and understand more...or wait for the developer who changed the behavior to see this and answer.

**Comment by ghost** [comment]
See this. https://github.com/laravel/framework/issues/52180#issuecomment-2236055605 It seems it is expected nehavior acc. to documentation.

**Comment by hafezdivandari** [comment]
This is expected behavior [since 10.x](https://laravel.com/docs/10.x/migrations#modifying-columns). You may check PR #45487 that explains the reason behind this. Also explained on [11.x upgrade guide](https://laravel.com/docs/11.x/upgrade#modifying-columns).
